### ISSUE TITLE
IDISP004 should not warn when directly disposing

### ISSUE THREAD

**Issue by chris579** [DotNetAnalyzers/IDisposableAnalyzers#136]
When directly disposing instances implementing `IDisposable` I get an IDISP004 warning.

This happens when writing 
```csharp
File.Create(path).Dispose();
```
and also when writing
```csharp
File.Create(path).Close();
```

It shouldn't warn in both cases.

**Comment by JohanLarsson** [comment]
Thanks for reporting!

I'm not sure about teaching the analyzer that `Close()` calls `Dispose()`

**Comment by JohanLarsson** [comment]
Specialcasing `Close()` would not be hard but I'm thinking maybe we want the analyzer a bit opinionated forcing us to write nice and uniform code 😄 

For the general case we can only really analyze code, decompiling and analyzing binaries would get messy I think.

**Comment by chris579** [comment]
I like the idea of forcing best practices. This way one wouldn’t have to look up the sources if method xy is really disposing the instance.
